### PR TITLE
Bug 1216084 - Fix the Gaia UI tests that are failing because of the new Homescreen

### PR DESCRIPTION
**Pull request by mwargers**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3722943669c08a05f95c1a2fba531dc1384e659a

File: tests/python/gaia-ui-tests/gaiatest/apps/base.py [hunk 1]
```diff
@@ -120,13 +120,16 @@ def keyboard(self):
...
 
+    @property
+    def is_displayed(self):
+        return (self.apps.displayed_app.manifest_url == self.manifest_url and
+               self.apps.displayed_app.entry_point == self.entry_point)
+
     def wait_to_be_displayed(self):
-        Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url == self.manifest_url and
-                                              self.apps.displayed_app.entry_point == self.entry_point)
+        Wait(self.marionette).until(lambda m: self.is_displayed == True)
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/base.py]: Good idea!

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py]: The removal the rocketbar page class is not done here. I guess that's because we want to keep them for the testing the old homescreen, right?

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+import urllib
+
+from marionette_driver import expected, By, Wait
+
+from gaiatest.apps.base import Base
+from gaiatest.apps.base import PageRegion
+
+
+class SearchPanel(Base):
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py]: Having a `Base` page class under the `regions` folder looks weird. I'd put a comment to say that it's temporary

File: tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py [hunk 4]
```diff
@@ -14,7 +14,6 @@
 class Homescreen(Base):
 
     name = 'Default Home Screen'
-    manifest_url = '{}verticalhome{}/manifest.webapp'.format(Base.DEFAULT_PROTOCOL,Base.DEFAULT_APP_HOSTNAME)
```
Comment (npark-mozilla) [inline tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py]: Just to confirm, even with the manifest_url gone, it'll still work with non-English locale right?

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py (hunk 2 above)
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py]: Following up from the discussion on IRC: we can delete the other rocket bar page class, for now

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        @property
+        def name(self):
+            return self.root_element.text
+
+        def tap(self):
+            app_name = self.name
+
+            self.root_element.tap()
+            # Wait for the displayed app to be that we have tapped
+            Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == app_name)
```
Comment (npark-mozilla) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py]: Again, would it work with non-English locale?

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py (hunk 2 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py]: Well, the whole homescreen app needs a revamp, but sure, I can remove the search_panel.py from the homescreen app.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py (hunk 3 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py]: I just copied this file from the homescreen folder, I think we have quite a few more of these files under regions, that use the `Base` page class.
I can add a comment that it is temporary, but should I mention a bug comment or something?

File: tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py (hunk 4 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py]: Yes, this is not related to locales at all.

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py (hunk 5 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py]: Probably not, I haven't investigated it.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e2f5445cafbdd232d4773e7b195ef08ef2f98c2f

File: tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py (hunk 3 above)
Comment (mwargers) [inline tests/python/gaia-ui-tests/gaiatest/apps/system/regions/search_panel.py]: Ok, I added a bug number here.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e75e6d008f2814ce4e9ae6067a85c6e63ecc809a
